### ISSUE TITLE
Several usability issues as of the last major UI Overhaul

### ISSUE THREAD

**Issue by adroslice** [nextcloud/notes#1062]
**Is your feature request related to a problem? Please describe.**
There are two problems I've encountered after recently performing 2 major upgrades on my selfhosted nextcloud instance, that are hindering my productivity. There is one other problem I'll describe however, that I believe has always been an issue.
1. Great productivity loss at narrow viewports. It is no longer possible to view both the notes list as well as the editing area when doing something like half-windowing on a 1080p screen at 100% scaling. You now have to navigate back and forth between your notes and the list, which means an extra click and, to use some design language, a loss of context. An added deficit is that even once you make it to the list, you still can't view your categories without popping over another sidebar.
2. Inability to rename a note in the list. This is probably the biggest deal-breaker. I never have the details window open, and even though it stays open as you open different notes, I find it wastes far too much space that could be used for the list or the note itself and doesn't fit the "minimal" and "distraction-free" design. 
3. Inability to right click to bring up a notes context menu. This was always missing, and clicking the little meatballs to get the rename item wasn't a huge deal. Out of all the problems mentioned in this issue, this is by far the smallest.

**Describe the solution you'd like**
1. A complete overhaul of the layout, or at least a reversion to how it used to be, or an option to use the old layout. I'm not saying the old one was perfect by any means, but the new version has made it strictly worse to use.
2. Just add back the context menu item, or always display the note title at the top to be edited. Honestly, that would be best.
3. Implement the right click app, if it's even a good implementation. If not, roll your own.

**Describe alternatives you've considered**
If these things aren't fixed, I see no real way for myself to mitigate or circumvent these issues, and would probably have to switch to using something else. Tiled window workflows are far too important for my own productivity, and renaming notes is something I do quite often.

**Comment by Yiannis128** [comment]
I agree that there are some usability issues, but I disagree on what needs to be done. Here's what I propose:
1. The 3 pane layout is essential due to the fact that the old layout was, quite simply, very confusing. However, it would be a nice QOL update if the notes list could be dragged so that the size would become zero. The draggable area could allow for the resizing to make it appear again.
2. I agree on the right click app being implemented, this is something that I was surprised was not already in, as most notes web apps have some form of right click functionality.

For the draggable notes list area, the notes list could have a minimum size like it has right now, but when you attempt to drag it to a smaller size than it already is, it snaps to 0, and then allows for more screen real estate for the note.
